### ISSUE TITLE
(Extremely often) on OSx I must press "CTRL + C" 

### ISSUE THREAD

**Issue by Swiffers** [KillianLucas/open-interpreter#561]
### Describe the bug

hi,

TL;DR: (Extremely often) on osx I must press "CTRL + C" because the executed program is stuck. 

I'm using the open-interpreter since it had 3000 stars, and this issue has always been here. I thought that it would be fixed after a few weeks of developments because it's very painful. 

The issue: When I run the open interpreter with -y (always the latest version), on average every ~15 commands it get stuck and I must press "CTRL + C" to send a signal. Sometimes I must press it multiple times. 

I would love to fix this issue, if anyone can help me here: which file should I look into ? I would like to create a fix that "when the program is stuck for X seconds" it automatically sends "CTRL + C" signal to continue. 

I can give as many details as needed - record a loom - just ask, many thanks.

### Reproduce

Run the open interpreter with -y on a macbook pro osx
Ask it run execute commands only

### Expected behavior

Continue working, but not being stuck indefinitely until I press CTRL + C (one or multiple times) depending on the situation. I have to be careful to not kill the prompt. 

### Open Interpreter version

0.1.7

### Python version

Python 3.11.2

### Operating System name and version

macOS 13.4 (22F66)

**Comment by ericrallen** [comment]
Hey there, @Swiffers!

I _think_ we might have some fixes for this coming out in an upcoming release.

**Comment by MikeBirdTech** [comment]
Closing this stale issue. Please create a new issue if the problem is not resolved or explained in the documentation. Thanks!

**Comment by csaroff** [comment]
This was still happening for me. No idea what the root-cause was, but restarting my computer fixed the problem.
